src/ed.cpp

```diff
@@ -388,17 +388,35 @@ bool srvSimpleQuery(ed_msgs::SimpleQuery::Request& req, ed_msgs::SimpleQuery::Re
             }
         }
 
-        bool geom_ok = true;
-        if (radius > 0)
+        if (radius < std::numeric_limits<double>::infinity())
         {
-            geom_ok = (radius * radius > (e->pose().t - center_point).length2());
-        }
+            bool geom_ok = false;
+            geo::ShapeConstPtr shape = e->shape();
+            if (radius > 0)
+            {
+                if (shape)
+                {
+                    geo::Vector3 center_point_e = e->pose().getBasis().transpose() * (center_point - e->pose().getOrigin());
+                    geom_ok = shape->intersect(center_point_e, radius);
+                }
+                else
+                {
+                    geom_ok = (radius * radius > (e->pose().t - center_point).length2());
+                }
+            }
+            else if (shape)
+            {
+                geo::Vector3 center_point_e = e->pose().getBasis().transpose() * (center_point - e->pose().getOrigin());
+                geom_ok = shape->contains(center_point_e);
+            }
```

Comment:
Would the code be less complex if first you do `if (shape)` and then do `if (radius > 0)`?

Reply:
No

Reply:
I will take another look at it today.

Reply:
See 89f4b08509d7108fbb1a28a49fb49c1a2f12718f